File: src/main/java/me/coderfrish/toml/reader/TomlParser.java
package me.coderfrish.toml.reader;

import me.coderfrish.toml.reader.node.Node;
import me.coderfrish.toml.reader.token.Token;

import java.util.List;

public class TomlParser {
    private final List<Token> tokens;
    private int offset = 0;

    public TomlParser(TomlLexer lexer) {
        this.tokens = lexer.tokenize();
    }

    public List<Node> parse() {
        return List.of();
    }
}

File: src/main/kotlin/me/coderfrish/toml/reader/TomlParser.kt
package me.coderfrish.toml.reader

import me.coderfrish.toml.api.Lexer
import me.coderfrish.toml.base.BaseParser
import me.coderfrish.toml.tokens.TokenType

class TomlParser(lexer: Lexer): BaseParser(lexer) {
    override fun parseIdentifier() {
        val key = peek().value
        advance()

        if (peek().type == TokenType.EQUALS) {
            advance()

            val value = peek().value
            println("key: $key, value: $value")
        }

        advance()
    }
}
